nibabel/freesurfer/io.py

```diff
@@ -129,6 +181,22 @@ def write_geometry(filepath, coords, faces, create_stamp=None):
         coords.astype('>f4').reshape(-1).tofile(fobj)
         faces.astype('>i4').reshape(-1).tofile(fobj)
 
+        # Add volume info, if given
+        if volume_info is None or len(volume_info) == 0:
+            return
+
+        for key, val in volume_info.items():
+            if key == 'head':
```

Comment:
Or, perhaps better, iterate through the original list (`for key in ['valid', 'filename', 'volume', 'voxelsize', 'xras', 'yras', 'zras', 'cras']`). This way we don't depend on the ordering of keys, if for some reason somebody wants to create their own `volume_info` dict.

I also think a strategy for missing values is a good idea. I'd be inclined to refuse to write an invalid chunk (since we'll fail to read it), so I would build the string and only write it on success. Perhaps this chunk could be re-worked into a sub-function like `_serialize_volume_info` that returns a string to write (or `b''` on failure, empty `volume_info`).